lib/sensors/temperature.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "i2c_sensors.hpp"
+
+#include <cstdint>
+#include <optional>
+
+#include <core/logger.hpp>
+#include <io/i2c.hpp>
+
+namespace hyped::sensors {
+
+static constexpr std::uint8_t kCtrl   = 0x04;
+static constexpr std::uint8_t kDataTH = 0x07;
+static constexpr std::uint8_t kDataTL = 0x06;
+static constexpr std::uint8_t kStatus = 0x05;
+// Sets the sensor to continuous mode, sets IF_ADD_INC, and sets sampling rate to 200Hz
+static constexpr std::uint8_t kConfigurationSetting = 0x3c;
+
+class Temperature : public II2cMuxSensor<std::int16_t> {
+ public:
+  Temperature(const std::uint8_t device_address,
+              const std::uint8_t channel,
+              io::I2c &i2c,
+              hyped::core::ILogger &log);
```

Comment:
order the arguments in the constructor the same way you ordered the class members